addons/mail/static/src/core/common/record.js

```diff
@@ -56,13 +56,13 @@ function updateAttr(record, fieldName, value) {
         if (!(value instanceof luxon.DateTime)) {
             value = deserializeDateTime(value);
         }
-        shouldChange = !record[fieldName]?.equals(value);
+        shouldChange = !(record[fieldName] && value.equals(record[fieldName]));
     }
     if (fieldDefinition?.type === "date" && value) {
         if (!(value instanceof luxon.DateTime)) {
             value = deserializeDate(value);
         }
-        shouldChange = !record[fieldName]?.equals(value);
+        shouldChange = !(record[fieldName] && value.equals(record[fieldName]));
```

Comment:
It would be nice to also add tests in records.js to cover the specific cases that made issues.